siteapp/views.py

```diff
@@ -1461,23 +1461,81 @@ def new_portfolio(request):
 @login_required
 def delete_portfolio(request, pk):
     """Form to delete portfolios"""
+
     if request.method == 'GET':
         portfolio = Portfolio.objects.get(pk=pk)
-        logger.info(
-            event="delete_portfolio",
-            object={"object": "portfolio", "id": portfolio.id, "title": portfolio.title},
-            user={"id": request.user.id, "username": request.user.username}
-        )
-        Portfolio.objects.get(pk=pk).delete()
-        return redirect("list_portfolios")
 
+        # Confirm user has permission to delete portfolio
+        CAN_DELETE_PORTFOLIO = False
+        if request.user.is_superuser:
+            CAN_DELETE_PORTFOLIO = True
+        if request.user.has_perm('delete_portfolio', portfolio):
+            CAN_DELETE_PORTFOLIO = True
```

Comment:
Super minor, but this could be combined into a single 'if' condition with an 'or'